### ISSUE TITLE
Can not merge or export recipes

### ISSUE THREAD

**Issue by ockham** [thinkle/gourmet#640]
Converted from [SourceForge issue 2961196](https://sourceforge.net/support/tracker.php?aid=2961196), submitted by [SourceForge user lillelund](https://sourceforge.net/users/?user_id=2825901) on 2010-03-01 13:34:52 UTC.

I have just installed the v0.14.5 version of Gourmet Recipe Manager.  I loaded a lot of MasterChef files into it, but noticed that it did not merge some of the recipes correctly.  I also had trouble exporting the recipe database.

I have done some debugging and found that there is a type mismatch in the convert.py module.  The procedure having trouble is the float_to_frac:

def float_to_frac (n, d=[2,3,4,5,6,8,10,16],approx=0.01,fractions=FRACTIONS_NORMAL):
    """Take a number -- or anything that can become a float --
    and attempt to return a fraction with a denominator in the list `d'. We
    approximate fractions to within approx. i.e. if approx=0.01, then 0.331=1/3"""
    if USE_FRACTIONS == FRACTIONS_OFF:
        return float_to_metric(n,approx)
    else:
        if not n: return ""
        n=float(n)
        i = int(n)
        if i >= 1:
            i="%s"%int(n)
        else:
            i=""
        rem = n - int(n)
        if rem==0 or rem<approx:
            if i:
                return "%i"%round(n)
            else:
                return "0"
        else:
            flag = False
            for div in d:
                f = fractify(rem,div,approx=approx,fractions=fractions)
                if f:
                    return " ".join([i,f]).strip()
             # use locale-specific metric formatting if fractions don't work
            return float_to_metric(n,approx)

The problem is that the round function does not return an integer, but instead returns a float.  To ensure that the conversion is done correctly the easiest is to wrap the round() function with an int():

def float_to_frac (n, d=[2,3,4,5,6,8,10,16],approx=0.01,fractions=FRACTIONS_NORMAL):
    """Take a number -- or anything that can become a float --
    and attempt to return a fraction with a denominator in the list `d'. We
    approximate fractions to within approx. i.e. if approx=0.01, then 0.331=1/3"""
    if USE_FRACTIONS == FRACTIONS_OFF:
        return float_to_metric(n,approx)
    else:
        if not n: return ""
        n=float(n)
        i = int(n)
        if i >= 1:
            i="%s"%int(n)
        else:
            i=""
        rem = n - int(n)
        if rem==0 or rem<approx:
            if i:
                return "%i"%int(round(n))
            else:
                return "0"
        else:
            flag = False
            for div in d:
                f = fractify(rem,div,approx=approx,fractions=fractions)
                if f:
                    return " ".join([i,f]).strip()
             # use locale-specific metric formatting if fractions don't work
            return float_to_metric(n,approx)

In other words, change line 873 in /usr/share/gourmet/gourmet/convert.py from "return "%i"%round(n)" to "return "%i"%int(round(n))" (without the quotes).

I have attached a patched and working version (on my machine - YMMV) of the convert.py

I hope this helps.

/Jan
